blocks/gnav/gnav.js

```diff
@@ -1,287 +1,414 @@
-        <div class="gnav-section"></div>
-        </div>
-        <div class="gnav-search"><div class="gnav-search-icon" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" id="gnav-search-icon" width="20" height="20" viewBox="0 0 24 24" focusable="false">
-            <path d="M14 2A8 8 0 0 0 7.4 14.5L2.4 19.4a1.5 1.5 0 0 0 2.1 2.1L9.5 16.6A8 8 0 1 0 14 2Zm0 14.1A6.1 6.1 0 1 1 20.1 10 6.1 6.1 0 0 1 14 16.1Z"></path></div>
-          </svg>
-          <div class="gnav-search-box-wrapper">
-            <div class="gnav-search-box gnav-nosearch">
-              <div class="gnav-search-input">
-                <input type="text" id="gnav-search-terms">
-                <svg xmlns="http://www.w3.org/2000/svg" id="gnav-search-icon" width="20" height="20" viewBox="0 0 24 24" focusable="false">
-                  <path d="M14 2A8 8 0 0 0 7.4 14.5L2.4 19.4a1.5 1.5 0 0 0 2.1 2.1L9.5 16.6A8 8 0 1 0 14 2Zm0 14.1A6.1 6.1 0 1 1 20.1 10 6.1 6.1 0 0 1 14 16.1Z"></path>
-                </svg>
-              </div>
-              <div id="gnav-search-results" class="gnav-search-results">
-              </div>
-              <div class="gnav-search-link">
-                <a href="${nav.search.href}">Try our advanced search</a>
-              </div>
-            </div>
-          </div>
-        </div>
-        <div class="gnav-signin"><a href="${nav.signIn.href}">${nav.signIn.text}</a></div>`;
-
-  gnav.innerHTML = html;
-
-  const hamburger = gnav.querySelector('.gnav-hamburger');
-  hamburger.addEventListener('click', () => {
-    const expanded = gnav.getAttribute('aria-expanded') === 'true';
-    gnav.setAttribute('aria-expanded', !expanded);
-  });
-
-  const search = gnav.querySelector('.gnav-search');
-  const searchIcon = gnav.querySelector('.gnav-search-icon');
-  const searchTerms = gnav.querySelector('#gnav-search-terms');
-
-  searchIcon.addEventListener('click', () => {
-    const expanded = (search.getAttribute('aria-expanded') === 'true');
-    collapseAll(gnav);
-    search.setAttribute('aria-expanded', !expanded);
-    if (!expanded) {
-      searchTerms.focus();
-    }
-  });
-
-  searchTerms.addEventListener('input', () => {
-    const searchResultsEl = gnav.querySelector('#gnav-search-results');
-    populateSearchResults(searchTerms.value, searchResultsEl);
-    const searchBox = gnav.querySelector('.gnav-search-box');
-    if (searchTerms.value.trim().length === 0) {
-      searchBox.classList.add('gnav-nosearch');
+    searchField.append(searchInput, advancedLink);
+    searchBar.append(searchField, searchResults);
+    return searchBar;
+  }
+
+  decorateProfile = () => {
+    const blockEl = this.body.querySelector('.profile');
+    if (!blockEl) return null;
+    const profileEl = createEl({ tag: 'div', className: 'gnav-profile' });
+    const envSuffix = this.env.ims === 'stg1' ? `-${this.env.ims}` : '';
+
+    window.adobeid = {
+      client_id: 'bizweb',
```

Comment:
You need to set `autoValidateToken: true` when working with imslib v2. This will ensure user state is preserved when navigating from one property to another.

Reply:
Fixed.